cpmpy/expressions/globalconstraints.py

```diff
@@ -463,19 +484,16 @@ def decompose_comparison(self, cpm_op, cpm_rhs):
             It is not a constraint itself, so it can not have a decompose().
             However, when used in a comparison relation: Element(arr,idx) <CMP_OP> CMP_RHS
             it is a constraint, and that one can be decomposed.
-            That is what this function does
-            (for now only used in transformations/reification.py)
+            Returns two lists of constraints:
+                1) constraint replacing node in expression tree
+                2) constraints which should be true at toplevel
+
         """
         from .python_builtins import any
 
         arr, idx = self.args
         return [(idx == i).implies(eval_comparison(cpm_op, arr[i], cpm_rhs)) for i in range(len(arr))] + \
-               [idx >= 0, idx < len(arr)]
-
-    def is_total(self):
-        arr, idx = self.args
-        lb, ub = get_bounds(idx)
-        return lb >= 0 & idx.ub < len(arr)
+               [idx >= 0, idx < len(arr)], []
```

Comment:
These are redundant now that Element is total.

Reply:
I've removed this new restriction to total, to be discussed in another PR e.g. #184